src/Microsoft.FeatureManagement/ConfigurationFeatureDefinitionProvider.cs

```diff
@@ -109,97 +137,77 @@ public Task<FeatureDefinition> GetFeatureDefinitionAsync(string featureName)
         public async IAsyncEnumerable<FeatureDefinition> GetAllFeatureDefinitionsAsync()
 #pragma warning restore CS1998
         {
-            EnsureInit();
-
             if (Interlocked.Exchange(ref _stale, 0) != 0)
             {
                 _definitions.Clear();
             }
 
-            //
-            // Iterate over all features registered in the system at initial invocation time
-            foreach (IConfigurationSection featureSection in GetFeatureDefinitionSections())
-            {
-                string featureName = GetFeatureName(featureSection);
+            IEnumerable<IConfigurationSection> microsoftFeatureManagementSection = GetFeatureDefinitionSections(MicrosoftFeatureManagementFields.FeatureManagementSectionName);
 
-                if (string.IsNullOrEmpty(featureName))
+            if (microsoftFeatureManagementSection != null)
```

Comment:
Usually IEnumerables don't come back as null, but rather empty. Is it really null?